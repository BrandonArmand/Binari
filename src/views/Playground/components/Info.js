import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Markdown from "react-markdown";
import PropTypes from "prop-types";
import { IconButton } from "@material-ui/core";

const useStyle = makeStyles({
    main: {
        background: "white",
        width: "100%",
        maxWidth: "100%",
        padding: "0 30px",
    },
});

const Info = props => {
    const classes = useStyle();
    const { text, } = props;

    return (
        <>
            <IconButton
                className="far fa-square"
                style={{
                    position: "absolute",
                    top: 0,
                    right: 30,
                    height:"20px",
                }}
                onClick={()=>props.show()}/>
            <IconButton
                className="fas fa-square"
                style={{
                    position: "absolute",
                    top: 0,
                    right: 60,
                    height:"20px",
                }}
                onClick={()=>props.expand()}/>
            <IconButton
                className="fas fa-window-minimize"
                style={{
                    position: "absolute",
                    top: 0,
                    right: 0,
                    height:"20px",
                }}
                onClick={()=>props.hide()}/>
            <Markdown className={classes.main} source={text} />
        </>
    );
};

Info.propTypes = {
    text: PropTypes.any,
    show: PropTypes.func,
    expand: PropTypes.func,
    hide: PropTypes.func,
};

export default Info;
