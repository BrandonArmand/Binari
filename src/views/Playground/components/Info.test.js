import React from "react"
import { render } from "@testing-library/react"
import Info from "./Info"
import jest  from "jest-mock"

describe("Info", () => {
    it("renders snapshot", () => {
        const props = {
            text: "Test",
            show: jest.fn(),
            expand: jest.fn(),
            hide: jest.fn(),
        }
        const { container } = render(<Info {...props} />)
        expect(container).toMatchSnapshot()
    })
})
