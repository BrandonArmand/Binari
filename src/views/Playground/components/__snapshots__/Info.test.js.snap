// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info renders snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiIconButton-root far fa-square"
    style="position: absolute; top: 0px; right: 30px; height: 20px;"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiIconButton-root fas fa-square"
    style="position: absolute; top: 0px; right: 60px; height: 20px;"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiIconButton-root fas fa-window-minimize"
    style="position: absolute; top: 0px; right: 0px; height: 20px;"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <div
    class="makeStyles-main-1"
  >
    <p>
      Test
    </p>
  </div>
</div>
`;
